ql/src/java/org/apache/hadoop/hive/ql/metadata/SessionHiveMetaStoreClient.java

```diff
@@ -321,7 +321,7 @@ public List<String> getTables(String dbName, String tablePattern) throws MetaExc
     if (tables == null || tables.size() == 0) {
       return tableNames;
     }
-    tablePattern = tablePattern.replaceAll("\\*", ".*");
+    tablePattern = tablePattern.replaceAll("\\.\\*", "\\*").replaceAll("\\*", ".*");
```

Comment:
i don't follow, you replace `.*` with `*` and then `*` with `.*` ?

Reply:
@wecharyu, could we replace it with `tablePattern.replaceAll("(?<!\\.)\\*", ".*")` ?